bin/compile

```diff
@@ -1,28 +1,95 @@
 #!/bin/bash
 
+set -e
+
 BUILD_DIR=$1
-#CACHE_DIR=$2
+CACHE_DIR=$2
 ENV_DIR=$3
 BUILDPACK_DIR="$(dirname "$(dirname "$0")")"
 
+function main() {
+  export_env_dir "$ENV_DIR"
+
+  # Get latest stable version from the tailscale pkgs page
+  TAILSCALE_VERSION=${TAILSCALE_VERSION:-$(curl --silent https://pkgs.tailscale.com/stable/ | \
+    awk '/option/ {gsub("<[^>]+>", ""); print $1}' | \
+    grep -v latest | sort -r | head -1)}
+
+  # if we have a cached copy of the tailscale executable...
+  if [ -f "$CACHE_DIR/tailscale" ]; then
```

Comment:
do we need to throw a `! -z "${CACHE_DIR}"` here or nah.

Reply:
Given that `CACHE_DIR` is passed as a required argument by the Heroku buildpack pipeline, I think it's probably safe to assume it's set. I also sorta think that the `-f <path>` would just return false if `CACHE_DIR` didn't exist... but I'd have to check that. Good thing I've got a test script to try it out with! haha 😄

Reply:
Yeah, confirmed that if the cache dir doesn't exist, -f will just return false. Seems fine to me. Though, it does bring up a question about whether or not we should check that we have a cache_dir before writing files into it... but again, I sorta think the cache_dir being passed as the second argument to this script should just be treated as part of the Heroku buildpack API contract.